autode/values.py

```diff
@@ -0,0 +1,203 @@
+import numpy as np
+from abc import ABC, abstractmethod
+from typing import Union
+from copy import deepcopy
+from autode.units import (Unit,
+                          ha, kjmol, kcalmol, ev,
+                          ang, a0, nm, pm, m,
+                          rad, deg)
+
+
+class Value(ABC):
+    """
+    Abstract base class for a value with a defined set of units, along with
+    perhaps other attributes and methods
+
+    x = Value(0.0)
+    """
+    implemented_units = []
+
+    def __repr__(self):
+        return self.__str__()
+
+    @abstractmethod
+    def __str__(self):
+        """String representation of this value"""
+
+    def copy(self):
+        return deepcopy(self)
+
+    def _convert_to_float(self, other):
+        """
+        Convert another value to these units, do nothing if not a Value
+
+        Arguments:
+            other (autode.values.Value | float):
+
+        Returns:
+            (float):
+        """
+
+        if not isinstance(other, Value):
+            return float(other)
+
+        return other.to(self.units).x
+
+    def __eq__(self, other):
+        """Equality of two values, which may be in different units
+        use default numpy close-ness to compare"""
+
+        if isinstance(other, Value):
+            if other.units == self.units:
+                return np.isclose(other.x, self.x)
+            else:
+                return np.isclose(other.to(self.units).x, self.x)
+
+        return np.isclose(other, self.x)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Less than comparison operator"""
+
+        if isinstance(other, Value):
+            return self.x < other.to(self.units).x
+
+        return self.x < other
+
+    def __gt__(self, other):
+        """Greater than comparison operator"""
+        return not self.__lt__(other)
+
+    def __le__(self, other):
+        """Greater than or equal to comparison operator"""
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __ge__(self, other):
+        """Less than or equal to comparison operator"""
+        return self.__gt__(other) or self.__eq__(other)
+
+    def __add__(self, other):
+        """Add another value onto this one"""
+        return self.__class__(self.x + self._convert_to_float(other),
```

Comment:
should you check for being a Value here?

Reply:
Should be checked in `_convert_to_float` (?)

Reply:
good point